cmd/devp2p/internal/ethtest/eth66_suite.go

```diff
@@ -356,12 +356,21 @@ func (s *Suite) TestMaliciousTx_66(t *utesting.T) {
 		invalidNonceTx(t, s),
 		hugeAmount(t, s),
 		hugeGasPrice(t, s),
+		hugeData(t, s),
 	}
-	t.Logf("Testing malicious tx propagation: sending old tx, invalid nonce tx, huge amount tx, huge gas price tx")
-	sendFailingTx66(t, s, badTxs)
-	// send huge data tx on its own to prevent node from rejecting tx announcement that's too large
-	t.Logf("Testing malicious tx propagation: sending huge data tx")
-	sendFailingTx66(t, s, []*types.Transaction{hugeData(t, s)})
+	sendConn := s.setupConnection66(t)
+	defer sendConn.Close()
+	for i, tx := range badTxs {
+		t.Logf("Testing malicious tx propagation: %v\n", i)
+		if err := sendConn.Write(&Transactions{tx}); err != nil {
+			t.Fatal(err)
+		}
+
+	}
+	recvConn := s.setupConnection66(t)
```

Comment:
The recvConn must be established before sending the txs, otherwise it may miss the announcement.